### PR TITLE
Replaced EntryTypes.getType('string') with direct access to the corre…

### PR DESCRIPTION
**Pull request by oscargus**
No more EntryTypes.getType("article"), instead BibtexEntryTypes.ARTICLE etc is directly accessed.

### CODE REVIEW

File: src/main/java/net/sf/jabref/importer/fileformat/SixpackImporter.java [hunk 1]
```diff
@@ -139,20 +141,18 @@ public boolean isRecognizedFormat(InputStream stream) throws IOException {
...
                     }
                     if (fields[1].equals("PhD-Thesis")) {
-                        type = "phdthesis";
+                        typ = BibtexEntryTypes.PHDTHESIS;
                     }
                     if (fields[1].equals("miscellaneous")) {
-                        type = "misc";
+                        typ = BibtexEntryTypes.MISC;
                     }
                     if (fields[1].equals("Conference")) {
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/importer/fileformat/SixpackImporter.java]: why not using `else if` in those checks?

**Comment by matthiasgeiger** [comment]
:+1:

File: src/main/java/net/sf/jabref/importer/fileformat/SixpackImporter.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/importer/fileformat/SixpackImporter.java]: Yeah, I asked myself that as well in the back of my head, totally focused on replacing stuff. :-)

But I'll change it now. Good point!

**Comment by simonharrer** [comment]
This might destroy biblatex support. As EntryTypes.getType may return a biblatex version if biblatex is activated.

**Comment by oscargus** [comment]
OK! So then it should be the other way around? Never use BibtexEntryTypes.*?

**Comment by simonharrer** [comment]
Hm, would make sense, yes. But there may be exceptions, probably. We should investigate this.

**Comment by koppor** [comment]
Damn it, we should really investigate it. We really like constants and it is common practice to use constants instead of strings. In the meantime, a comment should be added to BibtexEntryTypes.* to never use them in the code (and reference this pull request). :innocent:

**Pull request closed without merging**